### DISCUSSION TITLE
npm ERR! 401 Unauthorized - Unauthenticated: User cannot be authenticated with the token provided

### DISCUSSION THREAD

**Discussion by P91491** [community/community#40791] [Enterprise]
### Select Topic Area

Question

### Body

I am new to GitHub and I have created a ACTIONS in GitHub Enterprise 3.7 to publish the package to GitHub Package. But the job failed with log as below: 

![image](https://user-images.githubusercontent.com/118231094/205526742-221419b8-5ea3-4d6e-bb14-ef2d109949dc.png)

I have setup the Repository Token as shown below: enter image description here:
![image](https://user-images.githubusercontent.com/118231094/205526764-6df3ddf9-fcd2-4521-a846-27aca38f6ca3.png)

The value of Token I have retrieved from here enter image description here and assign to value for REPOSITORY secret NPM_TOKEN:
![image](https://user-images.githubusercontent.com/118231094/205526790-730fbda4-1a42-4b1d-8705-404ac2bed09e.png)
 

However, it just said the TOKEN cannot be authenticate. Any idea where I doing wrong?

***Add on - The ACTIONS will run success if I assign the TOKEN value where I get from my GitHub Access token (Setup in GitHub.com but not GitHub Enterprise), ACTION will run success. However, the published package will only show in my GitHub but not in my GitHub Enterprise. 
![image](https://user-images.githubusercontent.com/118231094/205526832-2a21080b-ec70-4236-8913-2e4d2dc26c25.png)

![image](https://user-images.githubusercontent.com/118231094/205526845-31e679f6-a885-4d4d-b470-1f189d68e3aa.png)

GitHub Enterprise Account will not able to see that Published Package. enter image description here
![image](https://user-images.githubusercontent.com/118231094/205526869-f0af341f-6720-430f-bd69-c413de6c3933.png)

**Comment by ossanna16** [comment]
Hi there @P91491 and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Reply by Giwan** [reply]
When setting up the action it looks like the default `baseUrl` points to 'https://api.github.com'. Using the enterprise server, means that the token is on your server, not GitHub's server. As a result the token is not recognized. 

I had the same issue while setting up an action on an enterprise server. 
This action from Peter Evans shows how to use the current server: 

```
this.octokit = new Octokit({
      auth: token,
      baseUrl: process.env['GITHUB_API_URL'] || 'https://api.github.com'
    })
```

[source: enable-automerge](https://github.com/peter-evans/enable-pull-request-automerge/blob/main/src/github-helper.ts)

When running the workflow action, the variable `process.env['GITHUB_API_URL']` is available.

What I like with this example is also that Peter shows how to provide the token in the defaults. This saves the consumer of the action from having to provide this value. 

```yml


token:
    description: 'GITHUB_TOKEN or a `repo` scoped Personal Access Token (PAT)'
    default: ${{ github.token }}

repository:
    description: 'The target GitHub repository containing the pull request'
    default: ${{ github.repository }}

```

https://github.com/peter-evans/enable-pull-request-automerge/blob/main/action.yml

Here's my custom action that works well both on on github.com as in an enterprise server environment.
